src/app/add-course-page/add-course-page/add-course-page.component.ts

```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
     this.authors = authors;
   }
 
-  updateCourse(): void {
-    this.course.creationDate = new Date(this.course.creationDate);
-    this.courseService.updateCourse(this.course);
+  async updateCourse(): Promise<void> {
+    await this.courseService.updateCourse(this.course);
+    this.router.navigate(['/courses']);
   }
 
-  newCourse(): void {
-    this.course.id = this.courseService.getCoursesList().length + 1 + ''
-    this.course.creationDate = new Date(this.course.creationDate);
+  async newCourse(): Promise<void> {
+    let courseData = await this.courseService.getAllCoursesList();
+    this.courses = courseData;
+
+    this.generateId();
+    this.course.id = this.temporaryId + '';
     this.courseService.addCourses(this.course);
+    this.router.navigate(['/courses']);
+  }
+
+  generateId(): void {
+    if (!this.courses.find(course => course.id === this.temporaryId + '')) {
+      return;
+    } else {
+      ++this.temporaryId;
+      this.generateId();
+    }
+  }
+
+  async takeCourseData(): Promise<void> {
```

Comment:
the same as updateCourse
and check all places where you used Promise<void> as return type and rewrite them